lessons/cryptography/cryptosystem-types/cryptosystems-types.md

```diff
@@ -0,0 +1,90 @@
+
+![Sending a Message with Symmetric Cryptography](.rsrc/symmetric-crypto.png)
+
+Alice encrypts a plaintext message with the shared private key, resulting in a ciphertext message. This encrypted ciphertext message is sent to Bob. Bob decrypts the ciphertext message with the shared private key, allowing him to recover the plaintext message sent by Alice. 
+
+### AES
+Advanced Encryption Standard (AES) is a type of symmetric cryptography, where the same private key is used to encrypt and decrypt a message. Currently, it would take billions of years to use a brute-force attack against the AES algorithm. However, a less sophisticated way that an attacker could break AES security is by stealing improperly protected private keys. For example, if Alice simply sent an unencrypted message to Bob with the private key so that they can both share it, an attacker might be able to intercept this message and thus be able to decrypt any future encrypted messages sent between Alice and Bob. 
+
+## Asymmetric Cryptosystems
+
+In an asymmetric cryptosystem, also known as a public-key cryptosystem, both the sender and receiver each have a private and a public key. Each person can generate their own private and public key pair. Each person should always keep their private key private, but can freely distribute their public key without any negative consequences. In the example below, Alice sends an encrypted message to Bob using an asymmetric cryptosystem. 
+
+![Sending a Message with Asymmetric Cryptography](.rsrc/asymmetric-crypto.png)
+
+In this example, Bob has shared his public key with Alice and keeps his private key private. Alice uses Bob's public key to encrypt a plaintext message, resulting in an encrypted ciphertext message. This encrypted message is sent to Bob. Bob then uses his private key to decrypt the ciphertext into Alice's original plaintext message. In this example, Alice's public and private keys are not needed since she is sending a message to Bob. If Bob wanted to send an encrypted message back to Alice, he would then need her public key, and Alice would use her private key to decrypt a message from Bob. 
+
+### RSA
+Rivest-Shamir-Adleman (RSA) is a type of asymmetric encryption that uses a private and public key based on two large prime numbers. The algorithm for encryption involves multiplying two very large prime numbers, and is considered very hard to crack because recovering the two original prime numbers is a mathematical problem that is currently thought to be infeasible to solve efficiently. There are some known attacks against RSA, but many of these attacks occur when small prime numbers are used. For example, if small prime numbers are used, an attacker could create a dictionary by encrypting high probability plaintext messages using various keys, and compare the resulting ciphertext with a stolen ciphertext message. If the messages match, the attacker now knows the original prime numbers used to encrypt the message. 
```

Comment:
the general attack on small primes is simple brute force factoring of the number and this is probably a better example than dictionary attacks

Reply:
Done